### PR TITLE
Documentation and code standards

### PR DESCRIPTION
**Pull request by googletorp**
Overall documentation and code standards improvements. This is mostly focusing on documentation, but some minor code changes have been made as well.

### CODE REVIEW

File: src/BulkDocs/BulkDocsInterface.php [hunk 1]
```diff
@@ -10,33 +10,53 @@
...
    * @param \Drupal\Core\Entity\ContentEntityInterface[] $entities
-   * @return \Drupal\relaxed\BulkDocs\BulkDocsInterface
+   *   Array of fully loaded content entities.
+   *
+   * @return $this
    */
   public function setEntities($entities);
 
   /**
+   * Gets the enties.
```
Comment (jeqq) [inline src/BulkDocs/BulkDocsInterface.php:49]: Here is a typo.

File: src/BulkDocs/BulkDocsInterface.php [hunk 2]
```diff
@@ -10,33 +10,53 @@
...
+   *
    * @return \Drupal\Core\Entity\ContentEntityInterface[]
+   *   Array of content entities.
    */
   public function getEntities();
 
   /**
-   * @return \Drupal\relaxed\BulkDocs\BulkDocsInterface
+   * Saves the enties on the bulk document.
```
Comment (jeqq) [inline src/BulkDocs/BulkDocsInterface.php:57]: ... here we have the same typo.

File: src/Changes/Changes.php [hunk 3]
```diff
@@ -55,10 +55,16 @@ static public function createInstance(ContainerInterface $container, SequenceInd
   }
 
   /**
+   * Construcs a BuldDocs object.
```
Comment (jeqq) [inline src/Changes/Changes.php:58]: It should be 'Constructs a Changes object'.

File: src/Changes/ChangesInterface.php [hunk 4]
```diff
@@ -17,32 +17,43 @@
...
+   *   The sequence to set.
+   *
+   * @return $this
    */
-  public function lastSeq($seq);
+  public function setLastSeq($seq);
 
   /**
-   * Return the changes in a 'normal' way.
+   * Get the changes in a 'normal' way.
```
Comment (jeqq) [inline src/Changes/ChangesInterface.php:51]: Let's use 'Gets' instead of 'Get' and 'Returns'.

File: src/Changes/ChangesInterface.php [hunk 5]
```diff
@@ -17,32 +17,43 @@
...
    */
   public function getNormal();
 
   /**
-   * Return the changes with a 'longpoll'.
+   * Get the changes with a 'longpoll'.
```
Comment (jeqq) [inline src/Changes/ChangesInterface.php:56]: Here the same, use 'Gets'.

File: src/Controller/ResourceController.php [hunk 6]
```diff
@@ -55,6 +59,8 @@ protected function getMethod() {
   }
 
   /**
+   * Get the request format.
```
Comment (jeqq) [inline src/Controller/ResourceController.php:62]: And here - use 'Gets'.

File: src/HttpMultipart/HttpFoundation/MultipartResponse.php [hunk 7]
```diff
@@ -5,8 +5,7 @@
 use Symfony\Component\HttpFoundation\Request;
 use Symfony\Component\HttpFoundation\Response;
 
-class MultipartResponse extends Response
-{
+class MultipartResponse extends Response {
```
Comment (jeqq) [inline src/HttpMultipart/HttpFoundation/MultipartResponse.php:8]: In the HttpMultipart directory we have a separate library - https://github.com/dickolsson/http-multipart. We'll probably remove it and import in other way. Let's keep the code standard as it is, for the moment.

File: src/Plugin/rest/resource/AttachmentResource.php [hunk 8]
```diff
@@ -43,7 +43,13 @@ class AttachmentResource extends ResourceBase {
    * @param string | \Drupal\file\FileInterface $file
    * @param string $scheme
    * @param string $filename
-   * @return ResourceResponse
+   *
+   * @throws \Symfony\Component\HttpKernel\Exception\AccessDeniedHttpException
+   *   Exception thrown if user doesn't have required access.
+   * @throws \Symfony\Component\HttpKernel\Exception\NotFoundHttpException
+   *   Exception thrown if $workspace or $entity is not loaded entities.
```
Comment (jeqq) [inline src/Plugin/rest/resource/AttachmentResource.php:50]: I think this sentence should be reformulated.

File: src/Plugin/rest/resource/AttachmentResource.php [hunk 9]
```diff
@@ -86,8 +93,18 @@ public function get($workspace, $entity, $field_name, $delta, $file, $scheme, $f
    * @param string $scheme
    * @param string $filename
    * @param \Drupal\file\FileInterface $received_file
+   *
+   * @throws \Symfony\Component\HttpKernel\Exception\AccessDeniedHttpException
+   *   Exception thrown if user doesn't have required access.
+   * @throws \Symfony\Component\HttpKernel\Exception\HttpException
+   *   Exception thrown file could not be saved.
+   * @throws \Symfony\Component\HttpKernel\Exception\NotFoundHttpException
+   *   Exception thrown if $workspace or $entity is not loaded entities.
```
Comment (jeqq) [inline src/Plugin/rest/resource/AttachmentResource.php:102]: And this.

**Comment by dickolsson** [comment]
There are also test failures in this pull request. But the failures might be related to changes in Drupal core.
@googletorp Can you please cherry-pick commit `3f5a339809dcc4b8ad81edb4b42ca87bc4a00677` into your PR here? This commit will update the .travis.yml file to test with a more stable version of Drupal core.

**Comment by jeqq** [comment]
This PR is outdated, closing it.

**Pull request closed without merging**